### PR TITLE
Fix: Building on Fedora, GCC 12.2.0

### PR DESCRIPTION
**Pull request by rohanverma94**
While building on Fedora with GCC 12.2, the CMake throws the following error in ```src/base/sequence_checker.cc``` & ```src/base/synchronization/auto_signaller.cc```

```
/libbase/src/base/sequence_checker.cc: In constructor ‘base::SequenceChecker::SequenceChecker(base::SequenceChecker&&)’:
/libbase/src/base/sequence_checker.cc:14:23: error: ‘exchange’ is not a member of ‘std’
   14 |   sequence_id_ = std::exchange(other.sequence_id_, std::nullopt);
      |                       ^~~~~~~~
/libbase/src/base/sequence_checker.cc: In member function ‘base::SequenceChecker& base::SequenceChecker::operator=(base::SequenceChecker&&)’:
/libbase/src/base/sequence_checker.cc:23:23: error: ‘exchange’ is not a member of ‘std’
   23 |   sequence_id_ = std::exchange(other.sequence_id_, std::nullopt);

/libbase/src/base/synchronization/auto_signaller.cc: In constructor ‘base::AutoSignaller::AutoSignaller(base::AutoSignaller&&)’:
/libbase/src/base/synchronization/auto_signaller.cc:16:19: error: ‘exchange’ is not a member of ‘std’
   16 |     : event_(std::exchange(other.event_, nullptr)) {}
      |                   ^~~~~~~~
/libbase/src/base/synchronization/auto_signaller.cc: In member function ‘base::AutoSignaller& base::AutoSignaller::operator=(base::AutoSignaller&&)’:
/libbase/src/base/synchronization/auto_signaller.cc:23:19: error: ‘exchange’ is not a member of ‘std’
   23 |     event_ = std::exchange(other.event_, nullptr);

```
This has been fixed by adding ```<utility>``` header in ```src/base/sequence_checker.h``` and ```src/base/synchronization/auto_signaller.h```.

### CODE REVIEW

File: src/base/sequence_checker.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
 #include <optional>
-
+#include <utility>
```
Comment (RippeR37) [inline src/base/sequence_checker.h:4]: 1. this should be in .cc file, not header, as the header doesn't need it
2. empty line between STL includes and library includes

File: src/base/synchronization/auto_signaller.h [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <cstddef>
+#include <utility>
```
Comment (RippeR37) [inline src/base/synchronization/auto_signaller.h:4]: move to .cc file, as the header file doesn't need it

**Review by RippeR37 — CHANGES_REQUESTED** [review]

File: src/base/sequence_checker.h (hunk 1 above)
Comment (rohanverma94) [reply src/base/sequence_checker.h]: Done with required changes.

File: src/base/synchronization/auto_signaller.h (hunk 2 above)
Comment (rohanverma94) [reply src/base/synchronization/auto_signaller.h]: Done with required changes.

File: src/base/sequence_checker.h [hunk 3]
```diff
@@ -1,7 +1,6 @@
 #pragma once
 
 #include <optional>
```
Comment (RippeR37) [inline src/base/sequence_checker.h:3]: Please revert this change/bring back the empty line between <optional> and logging.h includes here.

Also, squash everything into a single commit please.

**Review by RippeR37 — CHANGES_REQUESTED** [review]
